src/action.cpp

```diff
@@ -1019,47 +1014,57 @@ cata::optional<tripoint> choose_direction( const std::string &message, const boo
     return cata::nullopt;
 }
 
-cata::optional<tripoint> choose_adjacent( const std::string &message )
-{
-    return choose_adjacent( message, false );
-}
-
-cata::optional<tripoint> choose_adjacent( const std::string &message, const bool allow_vertical )
+cata::optional<tripoint> choose_adjacent( const std::string &message, bool allow_vertical )
 {
     const cata::optional<tripoint> dir = choose_direction( message, allow_vertical );
     return dir ? *dir + g->u.pos() : dir;
 }
 
 cata::optional<tripoint> choose_adjacent_highlight( const std::string &message,
-        const action_id action_to_highlight )
-{
-    return choose_adjacent_highlight( message, action_to_highlight, false );
-}
-
-cata::optional<tripoint> choose_adjacent_highlight( const std::string &message,
-        const action_id action_to_highlight, const bool allow_vertical )
+        const action_id action, bool allow_vertical )
 {
-    const std::function<bool( tripoint )> f = [&action_to_highlight]( tripoint p ) {
-        return can_interact_at( action_to_highlight, p );
+    const std::function<bool( tripoint )> f = [&action]( tripoint p ) {
+        return can_interact_at( action, p );
     };
     return choose_adjacent_highlight( message, f, allow_vertical );
 }
 
 cata::optional<tripoint> choose_adjacent_highlight( const std::string &message,
-        const std::function<bool ( tripoint )> &should_highlight, const bool allow_vertical )
+        const std::function<bool ( tripoint )> &allowed, bool allow_vertical, bool auto_select_if_single )
 {
     // Highlight nearby terrain according to the highlight function
-    bool highlighted = false;
-    for( const tripoint &pos : g->m.points_in_radius( g->u.pos(), 1 ) ) {
-        if( should_highlight( pos ) ) {
-            highlighted = true;
-            g->m.drawsq( g->w_terrain, g->u, pos,
-                         true, true, g->u.pos() + g->u.view_offset );
+    if( allowed != nullptr ) {
+        cata::optional<tripoint> single = cata::nullopt;
```

Comment:
```suggestion
        cata::optional<tripoint> single;
```
The default constructor creates an empty optional anyway.